torch/nn/modules/container.py

```diff
@@ -312,6 +325,29 @@ def __add__(self, other: Iterable[Module]) -> 'ModuleList':
             combined.add_module(str(i), module)
         return combined
 
+    def __repr__(self):
+        """A custom repr for ModuleList that compresses repeated module representations"""
+        list_of_reprs = [repr(item) for item in self]
+        repeats = [1]
+        repeated_blocks = [list_of_reprs[0]]
+        for r in list_of_reprs[1:]:
+            if r == repeated_blocks[-1]:
+                repeats[-1] += 1
+            else:
+                repeats.append(1)
+                repeated_blocks.append(r)
+
+        lines = []
+        main_str = self._get_name() + '('
+        for n, b in zip(repeats, repeated_blocks):
+            local_repr = f"{n} x {b}"
+            local_repr = _addindent(local_repr, 2)
+            lines.append(local_repr)
```

Comment:
I wonder if we want to preserve the key printing like we do for Modules. For example from:
```
ModuleList(
  (0): Linear(in_features=2, out_features=2, bias=True)
  (1): Linear(in_features=2, out_features=2, bias=True)
  (2): Linear(in_features=2, out_features=2, bias=True)
  (3): Conv2d(2, 2, kernel_size=(2, 2), stride=(1, 1))
)
```
to 
```
ModuleList(
  (0-2): 3 x Linear(in_features=2, out_features=2, bias=True)
  (3): Conv2d(2, 2, kernel_size=(2, 2), stride=(1, 1))
)
```

Reply:
This looks really good!

Reply:
Modified it. Here's an example output:


```
ModuleList(
  (0-4): 5 x Linear(in_features=10, out_features=11, bias=True)
  (5): Linear(in_features=11, out_features=10, bias=True)
  (6-15): 10 x LayerNorm((10,), eps=1e-05, elementwise_affine=True)
  (16): Linear(in_features=10, out_features=11, bias=True)
  (17): Linear(in_features=11, out_features=10, bias=True)
  (18-20): 3 x Linear(in_features=10, out_features=10, bias=True)
)
```